docs/controllers.md

```diff
@@ -0,0 +1,152 @@
+# Controllers
+
+## Introduction
+
+Forwarding controllers are the software components implementing the
+[`ControllerForwarding`](https://github.com/noble-assets/orbiter/blob/main/types/controller.go#L30-L33)
+interface responsible for handling forwarding packets.
+
+```go
+type ControllerForwarding interface {
+ Controller[core.ProtocolID]
+ PacketHandler[*ForwardingPacket]
+}
+```
+
+A [`ForwardingPacket`](https://github.com/noble-assets/orbiter/blob/main/types/packet.go#L171-L174)
+is composed by two parts:
+
+```go
+type ForwardingPacket struct {
+ TransferAttributes *TransferAttributes
+ Forwarding         *core.Forwarding
+}
+```
+
+- **Transfer attributes**: General information associated with the cross-chain transfer.
+- **Forwarding attributes**: Specific information associated with the protocol used to complete the
+  forwarding step.
+
+All controllers are orchestrated by the Orbiter component called
+[`Forwarder`](https://github.com/noble-assets/orbiter/blob/main/keeper/component/forwarder/forwarder.go#L42-L55).
+
+## Forwarding
+
+The forwarding step is completely defined by the core
+[`Forwarding`](https://github.com/noble-assets/orbiter/blob/main/types/core/orbiter.pb.go#L79-L91)
+type. This structure allows users to specify:
+
+1. The cross-chain protocol to use for the forwarding.
+2. The cross-chain protocol specific information.
+3. Passthrough metadata that has to be attached to the cross-chain transfer.
+
+The cross-chain protocol specific information is encoded with the `Attributes` field with an any
+type that implements the `ForwardingAttributes` interface. For additional details on the
+implementation, please refer to the
+[proto definition](https://github.com/noble-assets/orbiter/blob/main/proto/noble/orbiter/core/v1/orbiter.proto#L35-L55)
+
+Forwarding controllers are elements of the Orbiter module used to interpret the protocol-specific
+information and use them to forward funds. Based on the selected bridge, we can have:
+
+- **AutoCCTP**: Automatic forwarding of funds via the Cross-Chain Transfer Protocol.
+- **AutoLane**: Automatic forwarding of funds via the Hyperlane protocol.
+- **AutoIBC**: Automatic forwarding of funds via the Inter-BlockChain protocol.
+
+## Hyperlane
```

Comment:
If this PR is adding the details about the Hyperlane bridge, will there be another one for the specifics of CCTP? Because that implementation is already available as well so the docs should come too.